google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,507 @@ def run(
+                error_string = (
+                    "Invalid model ID. The model ID must be one of ["
+                    + ",".join(all_models)
+                    + "]. Note that deployed model IDs are different from the uploaded model's ID"
+                )
+                raise ValueError(error_string)
+            for key in objective_configs.keys():
+                if (
+                    model not in xai_enabled
+                    and objective_configs[key].explanation_config is not None
+                ):
+                    raise RuntimeError(
+                        "Invalid config for model ID %s. `explanation_config` should only be enabled if the model has `explanation_spec populated"
+                        % model
+                    )
+                all_configs.append(
+                    gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringObjectiveConfig(
+                        deployed_model_id=key,
+                        objective_config=objective_configs[key].as_proto(),
+                    )
+                )
+
+        return all_configs
+
+    @classmethod
+    def create(
+        cls,
+        endpoint: Union[str, "aiplatform.Endpoint"],
+        objective_configs: Optional[
+            Union[
+                model_monitoring.EndpointObjectiveConfig,
+                Dict[str, model_monitoring.EndpointObjectiveConfig],
+            ]
+        ] = None,
+        logging_sampling_strategy: Optional[model_monitoring.RandomSampleConfig] = None,
+        schedule_config: Optional[model_monitoring.ScheduleConfig] = None,
+        display_name: Optional[str] = None,
+        deployed_model_ids: Optional[List[str]] = None,
+        alert_config: Optional[model_monitoring.EmailAlertConfig] = None,
+        predict_instance_schema_uri: Optional[str] = None,
+        sample_predict_instance: Optional[str] = None,
+        analysis_instance_schema_uri: Optional[str] = None,
+        bigquery_tables_log_ttl: Optional[int] = None,
+        stats_anomalies_base_directory: Optional[str] = None,
+        enable_monitoring_pipeline_logs: Optional[bool] = None,
+        labels: Optional[Dict[str, str]] = None,
+        encryption_spec_key_name: Optional[str] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+        create_request_timeout: Optional[float] = None,
+    ) -> "ModelDeploymentMonitoringJob":
+        """Creates and launches a model monitoring job.
+
+        Args:
+            endpoint (Union[str, "aiplatform.Endpoint"]):
+                Required. Endpoint resource name or an instance of `aiplatform.Endpoint`. Format:
+                ``projects/{project}/locations/{location}/endpoints/{endpoint}``
+
+            objective_configs (Union[model_monitoring.objective.EndpointObjectiveConfig,
+                Optional. Dict[str, model_monitoring.objective.EndpointObjectiveConfig]):
```

Comment:
The type annotation in this part of the doc string looks incorrect.

Reply:
thanks for catching the typos! I've fixed it.